### PR TITLE
Move GitHub tasks to Celery worker

### PR DESCRIPTION
**Pull request by singingwolfboy**
Move GitHub tasks to a Celery worker

### CODE REVIEW

**Comment by singingwolfboy** [comment]
OK, I've verified that this branch runs successfully, and I can rescan repos successfully. @sarina @nedbat can you two review it now?

File: openedx_webhooks/github_views.py [hunk 1]
```diff
@@ -48,14 +47,24 @@ def pull_request():
...
         return "Ignoring synchronize events from github", 200
+    if event["action"] == "opened":
+        result = pull_request_opened.delay(pr, wsgi_environ=minimal_wsgi_environ())
+        status_url = url_for("tasks.status", task_id=result.id, _external=True)
+        resp = jsonify({"message": "queued", "status_url": status_url})
+        resp.status_code = 202
+        resp.headers["Location"] = status_url
+        return resp
+    if event["action"] == "closed":
+        result = pull_request_closed.delay(pr)
```
Comment (sarina) [inline openedx_webhooks/github_views.py]: This is the only line that seems to differ from the code under `if event["action"] == "opened"` so maybe,

```python
if event["action"] == "opened":
    result = pull_request_opened.delay(pr, wsgi_environ=minimal_wsgi_environ())
elif event["action"] == "closed":
    result = pull_request_closed.delay(pr)
else:
    # some error message? unknown action?
status_url = url_for("tasks.status", task_id=result.id, _external=True)
....
return resp
```

File: openedx_webhooks/tasks/github.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+        logger.info(msg)
+        return msg
+
+    repo = pr["base"]["repo"]["full_name"].decode('utf-8')
+    people = get_people_file(session=github)
+    custom_fields = get_jira_custom_fields(jira)
+
+    if user in people:
+        user_name = people[user].get("name", "")
+    else:
```
Comment (sarina) [inline openedx_webhooks/tasks/github.py]: Perhaps better as:
```python
user_name = people[user].get("name", "")
if not user_name:
    user_resp = github.get(pr["user"]["url"])
    ...
```

this way we don't end up with a blank user_name ever, at worst we get their github username.

**Comment by sarina** [comment]
Seems pretty reasonable to me

**Comment by singingwolfboy** [comment]
@sarina I responded to all your feedback -- thank you! @nedbat, do you have any suggestions?

**Pull request merged**